src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUrlBuilder.cs

```diff
@@ -113,9 +113,11 @@ private IEnumerable<string> ExpandGlobbedUrl(string include, string exclude)
                 if (!Cache.TryGetValue(cacheKey, out files))
                 {
                     var options = new MemoryCacheEntryOptions();
-                    foreach (var pattern in includePatterns)
+
+                    // Perf: Avoid allocating enumerator
+                    for (var i = 0; i < includePatterns.Length; i++)
                     {
-                        var changeToken = FileProvider.Watch(pattern);
+                        var changeToken = FileProvider.Watch(includePatterns[i]);
                         options.AddExpirationToken(changeToken);
```

Comment:
Only if the cache continually expires. So most of the time, no. Didn't feel like it reduced the readability of the code by changing it though. Maybe remove the comment?